include/pybind11/embed.h

```diff
@@ -149,6 +149,12 @@ inline void finalize_interpreter() {
 
     Py_Finalize();
 
+    // If something in Py_Finalize() calls get_internals() the pointer could be created, even if it
+    // didn't exist above.  (But we still use the above because if nothing calls get_internals() we
+    // might not have the static pointer set in get_internals_ptr())
+    if (!internals_ptr_ptr)
+        internals_ptr_ptr = &detail::get_internals_ptr();
```

Comment:
Good catch! I'd suggest moving up `get_internals_ptr()` to mirror the code from `detail:: get_internals()`. This is invariant with respect to `Py_Finalize()`:

```c++
auto *&internals_ptr = detail::get_internals_ptr();
if (builtins.contains(id) && isinstance<capsule>(builtins[id]))
    internals_ptr = *static_cast<internals **>(capsule(builtins[id]));

Py_Finalize();

delete internals_ptr;
internals_ptr = nullptr;
```